2025-07-18 12:14:56.401253+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 14:31:52.150141+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 14:31:59.661595+07:00 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/Andri/Belajar/Bahasa Pemrograman/sales-app/data/salesdb.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.util.JdbcUtils.getConnection(JdbcUtils.java:291)
	at org.h2.server.web.WebServer.getConnection(WebServer.java:811)
	at org.h2.server.web.WebApp.test(WebApp.java:977)
	at org.h2.server.web.WebApp.process(WebApp.java:241)
	at org.h2.server.web.WebApp.processRequest(WebApp.java:176)
	at org.h2.server.web.WebThread.process(WebThread.java:154)
	at org.h2.server.web.WebThread.run(WebThread.java:103)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/Andri/Belajar/Bahasa Pemrograman/sales-app/data/salesdb.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 22 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/Andri/Belajar/Bahasa Pemrograman/sales-app/data/salesdb.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 16 more
2025-07-18 14:58:01.702972+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 14:59:56.510813+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 15:12:03.192242+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 16:17:00.174329+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:10:31.348253+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:13:16.373796+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:13:27.854668+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:35:19.053454+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:38:40.102543+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:38:45.626154+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:40:33.917973+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 17:59:03.319122+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 19:20:34.866470+07:00 jdbc[13]: exception
org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
2025-07-18 19:23:00.167923+07:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "SELECT * FROM PRODUCT PRODUCTS [*]PENJUALAN"; SQL statement:
SELECT * FROM PRODUCT PRODUCTS PENJUALAN [42000-232]
